File: jansetu-ai/backend/ai/summary.py
def generate_summary(translated_text, category, department, urgency, location=None):
    key_issue = translated_text.split('.')[0].strip() if translated_text else "Complaint received"
    parts = [f"{category} complaint received.", f"Issue: {key_issue}."]
    if location:
        parts.append(f"Location: {location}.")
    urgency_map = {"critical": "requires immediate attention", "high": "requires urgent attention", "medium": "requires attention", "low": "routine matter"}
    parts.append(f"Urgency: {urgency} — {urgency_map.get(urgency, 'routine matter')}.")
    parts.append(f"Routed to: {department}.")
    return " ".join(parts)

File: jansetu-ai/core/summary.ts
// Port of backend/ai/summary.py

const URGENCY_DESCRIPTIONS: Record<string, string> = {
  critical: "requires immediate attention",
  high: "requires urgent attention",
  medium: "requires attention",
  low: "routine matter",
};

export function generateSummary(
  translatedText: string,
  category: string,
  department: string,
  urgency: string,
  location?: string
): string {
  const keyIssue = translatedText ? (translatedText.split(".")[0].trim() || "Complaint received") : "Complaint received";
  const parts: string[] = [`${category} complaint received.`, `Issue: ${keyIssue}.`];
  if (location) parts.push(`Location: ${location}.`);
  parts.push(`Urgency: ${urgency} — ${URGENCY_DESCRIPTIONS[urgency] ?? "routine matter"}.`);
  parts.push(`Routed to: ${department}.`);
  return parts.join(" ");
}
